src/controllers/hooks.js

```diff
@@ -93,8 +96,97 @@ async function verifyHelixSite(url) {
   return true;
 }
 
-async function extractDomainFromXForwardedHostHeader(forwardedHost) {
-  return forwardedHost.split(',')[0]?.trim(); // get the domain from x-fw-host header
+function parseHlxRSO(domain) {
+  const regex = /^([\w-]+)--([\w-]+)--([\w-]+)\.(hlx\.live|aem\.live)$/;
```

Comment:
a comment would've been nice here to explain the regex for the rest of ~~us~~ them mortals